src/components/Button/Button.stories.tsx

```diff
@@ -0,0 +1,198 @@
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { ReactNode } from "react";
+import { Button, ButtonProps } from "./Button";
+import { Box } from "../Box";
+import { RemoveIcon, SearchIcon } from "../Icons";
+import { Text } from "../Text";
+
+export default {
+  component: Button,
+} as ComponentMeta<typeof Button>;
+
+const ButtonSetWrapper = ({ children }: { children: ReactNode }) => {
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      placeItems="center"
+      gap={4}
+      marginX={6}
+    >
+      {children}
+    </Box>
+  );
+};
+
+const TextButtonVariants = (props: Omit<ButtonProps, "children">) => (
+  <>
+    <ButtonSetWrapper>
+      <Button variant="primary" size="lg" {...props}>
+        <Text variant="button" size="large" color="inherit">
```

Comment:
Text component is not intended to be used inside of the button it's rather for headers, section infos, messages etc. use text directly and use size scale. You actually can use it here as it renders "span" as well... but using text directly is just more natural here